workbench/planning/reporting.py

```diff
@@ -94,6 +98,52 @@ def add_planned_work_and_milestones(self, planned_work_qs, milestones_qs):
                 }
             )
 
+            absences_per_project = Absence.objects.filter(
```

Comment:
Uff, diese Abfrage einmal pro geplante Arbeit. Wenn Du eine Ansicht aufrufst mit vielen Arbeitseinheiten wird das super-teuer.

Gibt es keinen schlauen Weg, diese Abfrage aus dem äusseren `for`-Loop zu entfernen?

Reply:
Man könnte die `add_absences` methode verwenden um das zu optimieren. Die müsste man pro Projekt aufschlüsseln. Das wäre recht günstig.

Andererseits: Wenn man nur die Abwesenheiten filtern möchte, die mit den Zeiträumen der Planung konkurrieren. Vielleicht gibt's da auch eine bessere query.

Reply:
Ich schau mir das nochmal an. Ich hab ne Idee.